### PR TITLE
[WIP][TASK] Add coveralls to repo

### PR DESCRIPTION
**Pull request by SignpostMarv**
This commit adds coveralls.io config & coverage uploading tasks.
Coverage is listed last in travis so that coverage should only be
updated if a build passes.

regarding: #807, https://github.com/MyIntervals/emogrifier/pull/804#issuecomment-539230157

### CODE REVIEW

**Comment by SignpostMarv** [comment]
need to make coverage checks conditional, probably only running on latest-supported php version without `--prefer-lowest`, since the build fails with `--prefer-lowest` and running coverage where there's no version-specific conditionals is a little bit overkill.

* [first semi-successful build](https://travis-ci.org/SignpostMarv/emogrifier/builds/594853212)
* [coverage for reference](https://coveralls.io/builds/26167987)

**Comment by oliverklee** [comment]
To speed up the build (and not hog Travis Ci resources more than necessary), I'd like to keep xdebug deactivated unless we really need it.

Can we have something like this? "collect the coverage (and deactivate xdebug) only on the highest supported PHP version and maybe only with the latest dependencies"

**Comment by oliverklee** [comment]
`.coveralls.yml' should also be marked as "don't export" in `.gitattributes` so this file will not be packaged.

**Comment by SignpostMarv** [comment]
> `.coveralls.yml` should also be marked as "don't export" in `.gitattributes` so this file will not be packaged.

resolved in 141559d

**Comment by SignpostMarv** [comment]
5aa418d will fail as I'm just checking the syntax on the conditional unloading of xdebug.

File: .gitattributes [hunk 1]
```diff
@@ -6,3 +6,4 @@
 /config/ export-ignore
 /phpcs.xml.dist export-ignore
 /tests/ export-ignore
+/.coveralls.yml export-ignore
```
[the remark this answers is not in the document]
Comment (SignpostMarv) [reply .gitattributes]: got that ready, just waiting to see how this latest build goes so I don't have to rebase behind it.

[the remark this answers is not in the document]
Comment (SignpostMarv) [reply .gitattributes]: resolved in 8a6d046

**Comment by SignpostMarv** [comment]
I think the PR build is failing because it was done in advance of #807 task 2

**Comment by oliverklee** [comment]
> I think the PR build is failing because it was done in advance of #807 task 2

I'll look into it.

**Comment by SignpostMarv** [comment]
for reference:
* https://travis-ci.org/SignpostMarv/emogrifier/builds/595265009
* https://coveralls.io/builds/26188572

**Comment by SignpostMarv** [comment]
> I'll look into it.

I'm not sure if retried PR travis builds update the status of PRs, so I can add the #807 task 3 badge to trigger a fresh build (which I think is preferable to to merging the PR without a green tick to force a fresh build of the master branch)

**Comment by JakeQZ** [comment]
> I'm not sure if retried PR travis builds update the status of PRs

AFAIAA it does.  Occasionally Travis does error for no apparent reason, and restarting the jobs does resolve this.  Though if there's a genuine error, perhaps due something going awry with a merge/rebase, it won't go away.

> preferable to to merging the PR without a green tick

I don't think the system will allow us to merge to `master` without a green tick.  At least, that's the way I think it's supposed to work :)

**Comment by JakeQZ** [comment]
> > I think the PR build is failing because it was done in advance of #807 task 2
> 
> I'll look into it.

@oliverklee, with the recent changes you've made in this area, should the build failure be resolved now if this branch is rebased?

**Comment by SignpostMarv** [comment]
> should the build failure be resolved now if this branch is rebased?

the build failure was due to coveralls.io not being set up for this repo, iirc.

**Comment by oliverklee** [comment]
I will switch the default git branch from `master` to `main` in a minute. GitHub then will autoclose all PRs targeted at the removed/renamed branch. Please create a new PR (from the existing PR branch) targeting `main`.

**Pull request closed without merging**